### PR TITLE
Create Resources folder in the Extensions directory

### PR DESCRIPTION
**Pull request by n-boshnakov**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR:
- adds an Resources folder to the Extensions directory
- moves topics with "Resources" in the title to the new folder
- fixes broken links due to move
- minor style fixes
- minor naming convention fixes for other files

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```doc developer
NONE
```

### CODE REVIEW

File: docs/README.md
Comment (rfranzke) [inline docs/README.md]: Why this stuttering/duplication in the folder name? Remove the `extension-` prefix?

Comment (n-boshnakov) [reply docs/README.md]: The naming was suggested by Viktor Kostov (@Kostov6). I am not opposed to removing the stuttering, but perhaps he can say if there is a specific purpose to it first.

Comment (Kostov6) [reply docs/README.md]: Honestly I was split between `resources` and `extension-resources`. I agree removing it would be better.

Comment (n-boshnakov) [reply docs/README.md]: Alright. In that case, I will change the name of the folder to `resources`.

File: docs/extensions/resources/_index.md
Comment (Kostov6) [inline docs/extensions/resources/_index.md]: Let's create this empty `_index.md` through the manifest

File: docs/README.md
Comment (ialidzhikov) [inline docs/README.md]: Broken link

File: docs/extensions/resources/dns-record.md
Comment (ialidzhikov) [inline docs/extensions/resources/dns-record.md]: Revert the file name to `dnsrecord.md`.

File: docs/README.md
Comment (ialidzhikov) [inline docs/README.md]: The Network resource is an extension resource as well. Why is it not moved?

File: docs/README.md
Comment (ialidzhikov) [inline docs/README.md]: The ContainerRuntime resource is an extension resource as well. Why is it not moved?

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: docs/extensions/cluster.md
Comment (ialidzhikov) [inline docs/extensions/cluster.md]:
```suggestion
These extensions often serve a well-defined purpose (e.g., the management of [DNS records](./resources/dnsrecord.md), [infrastructure](./resources/infrastructure.md)).
```

File: docs/extensions/cluster.md
Comment (ialidzhikov) [inline docs/extensions/cluster.md:7]: One more occurrence in this file:
```diff
diff --git a/docs/extensions/cluster.md b/docs/extensions/cluster.md
index b4576db7e0..798b87e6f7 100644
--- a/docs/extensions/cluster.md
+++ b/docs/extensions/cluster.md
@@ -52,7 +52,7 @@ There are some fields in the `Shoot` specification that might be interesting to

 In some cases, Gardener may create extension resources that are not associated with a shoot, but are needed to support some functionality internal to Gardener. Such resources will be created in the `garden` namespace of a seed cluster.

-For example, if the [managed ingress controller](../deployment/deploy_gardenlet_manually.md) is active on the seed, Gardener will create a [DNSRecord](dnsrecord.md) resource(s) in the `garden` namespace of the seed cluster for the ingress DNS record.
+For example, if the [managed ingress controller](../deployment/deploy_gardenlet_manually.md) is active on the seed, Gardener will create a [DNSRecord](./resources/dnsrecord.md) resource(s) in the `garden` namespace of the seed cluster for the ingress DNS record.

 Extension controllers that may be expected to reconcile extension resources in the `garden` namespace should make sure that they can tolerate the absence of a cluster resource. This means that they should not attempt to read the cluster resource in such cases, or if they do they should ignore the "not found" error.
```

File: docs/extensions/referenced-resources.md
Comment (ialidzhikov) [inline docs/extensions/referenced-resources.md:29]: Revert the file name change for `docs/extensions/managed-resources.md`.
Revert it for `docs/extensions/controller-registration.md` as well.

Comment (n-boshnakov) [reply docs/extensions/referenced-resources.md:29]: I agree for reverting the change to `controller-registration.md`, as the resource name is `ControllerRegistration`. 

However, I still think that `Referenced Resources` topic name should be split for readability, as it is essentially two separate words.

Comment (ialidzhikov) [reply docs/extensions/referenced-resources.md:29]: I updated the comment, it is rather for reverting the name for `docs/extensions/managed-resources.md` as I am not allowed to comment in the affected file.

File: docs/README.md
Comment (ialidzhikov) [reply docs/README.md]: This comment was marked as resolved but it is not resolved actually

```suggestion
    * [`DNSRecord` resources](extensions/resources/dnsrecord.md)
```

File: docs/README.md
Comment (ialidzhikov) [inline docs/README.md:187]: Most of the extension resources have a title. Example: https://github.com/gardener/gardener/blob/6e98ed02183da06434f957be724b24cb1d9b2e62/docs/extensions/resources/backupbucket.md#L1-L3

But `docs/extensions/resources/infrastructure.md`, `docs/extensions/resources/bastion.md`, `docs/extensions/resources/extension.md`, `docs/extensions/resources/network.md`, `docs/extensions/resources/operatingsystemconfig.md` and `docs/extensions/resources/worker.md` don't have such title. 

Please unify and make it consistent everywhere!

File: docs/README.md
Comment (ialidzhikov) [inline docs/README.md:199]: Please, for each moved file take care to find the usages outside of this repo and raise a PR for them.

For example the extension.md file is referenced by repos like [registry-cache](https://github.com/gardener/gardener-extension-registry-cache/blob/2f99e1fa965c8b8d33778fc82861889ba1e9a33f/docs/development/extension-registry-cache.md?plain=1#L18).

File: docs/extensions/resources/containerruntime.md
Comment (ialidzhikov) [inline docs/extensions/resources/containerruntime.md]: Broken link:

```suggestion
Gardener supports creation of Worker machines using CRI. For more information, see [CRI Support](./operatingsystemconfig.md#cri-support).
```

File: docs/README.md
Comment (ialidzhikov) [inline docs/README.md:195]: Broken link in https://github.com/gardener/gardener/blob/6ba5765c27a1cb90e0fcb11764af7cfe6cc013cd/pkg/component/extensions/operatingsystemconfig/nodeinit/nodeinit.go#L74

File: docs/usage/shoot/shoot_workerless.md
Comment (ialidzhikov) [inline docs/usage/shoot/shoot_workerless.md]: Broken link

```suggestion
 - extension components (if they support workerless `Shoot`s, see [here](../../extensions/resources/extension.md#what-is-required-to-register-and-support-an-extension-type))
```

File: docs/proposals/21-ipv6-singlestack-local.md
Comment (ialidzhikov) [inline docs/proposals/21-ipv6-singlestack-local.md]: Broken link, revert

```suggestion
      - [DNS Records](#dns-records)
```

File: docs/proposals/01-extensibility.md
Comment (ialidzhikov) [inline docs/proposals/01-extensibility.md]: Broken link, revert

```suggestion
        - [DNS Records](#dns-records)
```

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

**Review by Kostov6 — COMMENTED** [review]
/lgtm

File: docs/README.md
Comment (n-boshnakov) [reply docs/README.md:199]: I will do that once the PR gets merged and a list of all broken links is generated during the build.

File: docs/README.md
Comment (n-boshnakov) [reply docs/README.md:187]: The comment not being resolved could be an issue due to the rebase. I will go fix it.

Comment (n-boshnakov) [reply docs/README.md:187]: As for the titles, the ones that have a specific title set in the frontmatter have a unique capitalization that can't be reflected in their file name. If a topic does not have a title set in its frontmatter, its title will be its file name but capitalized -> `bastion.md` = Bastion

Therefore, for such topics it isn't needed to set a title in the frontmatter. I will set the title for `operatingsystemconfig.md`, though, as the title there should be OperatingSystemConfig

**Review by ialidzhikov — APPROVED** [review]
/lgtm

File: docs/extensions/resources/extension.md
Comment (ialidzhikov) [inline docs/extensions/resources/extension.md]:
```suggestion
Gardener creates Extension resources as part of the Shoot reconciliation. Moreover, it is guaranteed that the [Cluster](cluster.md) resource exists before the `Extension` resource is created. `Extension`s can be reconciled at different stages during Shoot reconciliation depending on the defined extension lifecycle strategy in the respective [ControllerRegistration](../controllerregistration.md) resource. Please consult the [Extension Lifecycle](../controllerregistration.md#extension-lifecycle) section for more information.
```

/hold

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/hold cancel

**Comment by ialidzhikov** [comment]
```
[FAILED] Timed out after 10.001s.
  Expected success, but got an error:
      <*errors.StatusError | 0xc0046bb400>: 
      servicemonitors.monitoring.coreos.com "garden-gardener-operator" not found
      {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "servicemonitors.monitoring.coreos.com \"garden-gardener-operator\" not found",
              Reason: "NotFound",
              Details: {
                  Name: "garden-gardener-operator",
                  Group: "monitoring.coreos.com",
                  Kind: "servicemonitors",
                  UID: "",
                  Causes: nil,
                  RetryAfterSeconds: 0,
              },
              Code: 404,
          },
      }
  In [It] at: /home/prow/go/src/github.com/gardener/gardener/test/integration/operator/garden/garden/garden_test.go:[431](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/10632/pull-gardener-integration/1849055781568122880#1:build-log.txt%3A431) @ 10/23/24 11:56:27.444
```

Occurrence of https://github.com/gardener/gardener/issues/10576

---

/test pull-gardener-integration

**Comment by ialidzhikov** [comment]
```
Starting deploy...
waiting for deletion: running [kubectl --context garden --kubeconfig /home/prow/go/src/github.com/gardener/gardener/example/operator/virtual-garden/kubeconfig get -f - --ignore-not-found -ojson]
 - stdout: ""
 - stderr: "Error from server (ServiceUnavailable): the server is currently unable to handle the request (get gardenlets.seedmanagement.gardener.cloud local)\n"
 - cause: exit status 1
make[2]: *** [Makefile:400: operator-seed-up] Error 1
make[2]: Leaving directory '/home/prow/go/src/github.com/gardener/gardener'
make[1]: *** [Makefile:424: test-e2e-local-operator-seed] Error 2
make[1]: Leaving directory '/home/prow/go/src/github.com/gardener/gardener'
```

/test pull-gardener-e2e-kind-operator-seed

**Pull request merged**
